### ISSUE TITLE
Data race in Shar.createFieldsAndMeasurements/Measurement.HasFields

### ISSUE THREAD

**Issue by jwilder** [influxdb/influxdb#3663]
```
WARNING: DATA RACE
Write by goroutine 271:
  runtime.mapassign1()
      /usr/local/go/src/runtime/hashmap.go:383 +0x0
  github.com/influxdb/influxdb/tsdb.(*Shard).createFieldsAndMeasurements()
      /home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/tsdb/shard.go:308 +0x5f4
  github.com/influxdb/influxdb/tsdb.(*Shard).WritePoints()
      /home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/tsdb/shard.go:167 +0x2c3
  github.com/influxdb/influxdb/tsdb.(*Store).WriteToShard()
      /home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/tsdb/store.go:286 +0x15e
  github.com/influxdb/influxdb/cluster.func·002()
      /home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/cluster/points_writer.go:250 +0x37f

Previous read by goroutine 190:
  runtime.mapaccess2_faststr()
      /usr/local/go/src/runtime/hashmap_fast.go:281 +0x0
  github.com/influxdb/influxdb/tsdb.(*Measurement).HasField()
      /home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/tsdb/meta.go:325 +0xcc
  github.com/influxdb/influxdb/tsdb.createTagSetsAndFields()
      /home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/tsdb/mapper.go:786 +0x236
  github.com/influxdb/influxdb/tsdb.(*LocalMapper).Open()
      /home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/tsdb/mapper.go:175 +0xdaa
  github.com/influxdb/influxdb/tsdb.(*Executor).executeAggregate()
      /home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/tsdb/executor.go:358 +0x630
  runtime.goexit()
      /usr/local/go/src/runtime/asm_amd64.s:2232 +0x0
  time.Time.date()
      /usr/local/go/src/time/time.go:674 +0x47
  time.Time.Date()
      /usr/local/go/src/time/time.go:291 +0x4c
  log.(*Logger).formatHeader()
      /usr/local/go/src/log/log.go:84 +0x2d8
  log.(*Logger).Output()
      /usr/local/go/src/log/log.go:148 +0x292
  log.(*Logger).Printf()
      /usr/local/go/src/log/log.go:160 +0x92
  github.com/hashicorp/raft.(*NetworkTransport).listen()
      /home/ubuntu/influxdb-build/src/github.com/hashicorp/raft/net_transport.go:351 +0x1e3

Goroutine 271 (running) created at:
  github.com/influxdb/influxdb/cluster.(*PointsWriter).writeToShard()
      /home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/cluster/points_writer.go:271 +0x42c
  github.com/influxdb/influxdb/cluster.func·001()
      /home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/cluster/points_writer.go:205 +0x149

Goroutine 190 (running) created at:
  github.com/influxdb/influxdb/tsdb.(*Executor).Execute()
      /home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/tsdb/executor.go:90 +0x147
  github.com/influxdb/influxdb/tsdb.(*QueryExecutor).executeSelectStatement()
      /home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/tsdb/query_executor.go:261 +0xba
  github.com/influxdb/influxdb/tsdb.func·008()
      /home/ubuntu/influxdb-build/src/github.com/influxdb/influxdb/tsdb/query_executor.go:146 +0x97a
```

**Comment by DanielMorsing** [comment]
Are the reproduction steps for this that same as in #3667?

**Comment by DanielMorsing** [comment]
closed by #3671
